[Accessibility] Make shipping address form labels visible on legacy Cart

Manussakis opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #51900 .

- Make the shipping address form labels visible on the legacy Cart page.

### Example of regressions in the shipping address form on Storefront and Astra themes

The regressions are caused by these themes not loading the Woo core stylesheet.
Removing the `screen-reader-text` from labels makes them visible, which changes the layout a bit.
This approach was proposed in section two of [this comment](https://github.com/woocommerce/woocommerce/pull/52272#issuecomment-2555712159).

- **Before – Storefront, Classic Cart**
    ![image](https://github.com/user-attachments/assets/f0a83f86-e942-4ded-b4f0-db92399ddf20)

- **After – Storefront, Classic Cart**
    ![image](https://github.com/user-attachments/assets/2164f5fe-3ab8-4b44-8459-adfac77a856d)

- **Before – Astra, Classic Cart**
    ![image](https://github.com/user-attachments/assets/c9e58130-6a8b-4e19-8224-3b455d081156)

- **After – Astra, Classic Cart**
    ![image](https://github.com/user-attachments/assets/c48f65e1-06e9-4b3a-b75b-80037418f4cc)
    - Removing the `screen-reader-text` class from labels makes them visible.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


#### Shipping address form
1. Make sure your store uses the classic Cart page.
2. Add a product to the cart and go to the Cart page.
3. Click on the "Change Address" link in the Shipping section to open the Shipping Address form:
    ![Screenshot 2025-01-03 at 17 02 16](https://github.com/user-attachments/assets/97ced173-8bc6-410c-b708-e4cef60031a1)
4. Verify the labels are above the inputs.
  ![Screenshot 2025-01-03 at 17 04 11](https://github.com/user-attachments/assets/d829399a-cf93-486b-a4fe-e58d67971657)
5. Perform the test on the WP default themes and Storefront.
6. Verify there is no placeholder inside the input.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Manussakis committed:
Create woocommerce float label mixin

Manussakis committed:
Create float label styles

Manussakis committed:
Create float label styles for TT3 theme

Manussakis committed:
Create float label functionality

Manussakis committed:
Add float label functionality to coupon input on legacy cart

Manussakis committed:
Add float label functionality to shipping calculator on legacy cart

Manussakis committed:
Update woocommerce float label mixin

Manussakis committed:
Add float label style to T17 theme

Manussakis committed:
Add float label style to T19 theme

Manussakis committed:
Add float label style to TT theme

Manussakis committed:
Add float label style to TT1 theme

Manussakis committed:
Add float label style to TT2 theme

Manussakis committed:
Add float label style to TT3 theme

Manussakis committed:
Update is-active float label style

Manussakis committed:
Add changelog file

Manussakis committed:
Replace getByPlaceholder with getByLabel in tests

Manussakis committed:
Bump cart template version

Manussakis committed:
Add is-active class to label if input is the current active element

Manussakis committed:
Add float label to coupon in legacy Checkout page

Manussakis committed:
Update changelog text

Manussakis commented on plugins/woocommerce/client/legacy/css/_mixins.scss:
```diff
@@ -383,3 +383,22 @@
 	}
 }
+
+@mixin woocommerce-float-label {
```
Create a mixin with the float label style to be used by the Twenty* themes stylesheets.

Manussakis commented on plugins/woocommerce/client/legacy/css/twenty-nineteen.scss:
```diff
@@ -33,19 +33,40 @@ a.button {
+	
+				label {
+					@include woocommerce-float-label();
+					left: 0.8em;
+					transform: translateY(0.4em);
```
Each Twenty* theme requires adjustments to better position the label inside the input.

Manussakis commented on plugins/woocommerce/client/legacy/css/twenty-nineteen.scss:
```diff
@@ -1034,6 +1073,40 @@ table.variations {
 }
 
+.woocommerce-shipping-totals {
+	.select2-container {
```
For non-block themes (Twenty Nineteen and Twenty Seventeen), add style to the `select2` inputs so they get the same appearance as the text inputs.

**Before**
![image](https://github.com/user-attachments/assets/64879b38-6c63-476f-8282-f06bef59511d)

**After**
![image](https://github.com/user-attachments/assets/2e6d7f77-4b01-4cc8-a524-b134057f1055)

Manussakis commented on plugins/woocommerce/client/legacy/css/_mixins.scss:
```diff
@@ -383,3 +383,22 @@
+	transform: translateY(0.9em);
+	transform-origin: center left;
+	transition: transform 0.2s ease;
+	z-index: 1;
```
Make sure the label is visible in front of the input if the input has a background color.

Manussakis commented on plugins/woocommerce/client/legacy/css/woocommerce-blocktheme.scss:
```diff
@@ -246,6 +246,10 @@ a.added_to_cart {
 			height: 3.5rem;
 		}
+		
+		.select2-selection--single .select2-selection__rendered {
+			padding-left: 0;
```
Better align the select value with the label of the `select2` input.
![Screenshot 2024-10-25 at 17 25 30](https://github.com/user-attachments/assets/64c17f40-b0c0-4f69-bdaa-9cfce568a65e)

Manussakis commented on plugins/woocommerce/client/legacy/js/frontend/country-select.js:
```diff
@@ -158,7 +158,7 @@ jQuery( function( $ ) {
 					.prop( 'id', input_id )
 					.prop( 'name', input_name )
-					.prop( 'placeholder', placeholder )
+					.prop( 'placeholder', input_classes.includes( 'input-with-wc-float-label' ) ? '' : placeholder )
```
If the input has a floating label, prevent the `placeholder` attribute from getting a value after updating the country select box in the shipping calculator.

Manussakis commented on plugins/woocommerce/client/legacy/js/frontend/woocommerce.js:
```diff
@@ -194,9 +196,54 @@ function refresh_sorted_by_live_region () {
+		var label = get_label_from_input_id( input.id );
+
+		if ( label ) {
+			var fake_placeholder = label.parentElement.querySelector( '[class*="placeholder"]' );
```
The `select2` library creates a fake placeholder like this:

```html
<span class="select2-selection__placeholder">Select an option…</span>
```
![Screenshot 2024-10-25 at 17 32 43](https://github.com/user-attachments/assets/0ef075c2-08d8-470f-881d-5f262ea3f34c)

In this case, the `label` element needs to get the `is-active` class so it doesn't overlap with the fake placeholder.

Manussakis commented:
@dkotter This task is ready for code review.

I left some comments that will help provide more context on the changes.

I noticed a warning that the `shipping-calculator.php` template needs to have its version bumped, however, [the version is already 9.5.0.](https://github.com/woocommerce/woocommerce/blob/e2ebc335f25df3969d2df0543b96d4d106368b22/plugins/woocommerce/templates/cart/shipping-calculator.php#L15)

Manussakis committed:
Adjust select value position in non-block themes

dkotter reviewed (approved):
@ankitguptaindia @qasumitbagthariya This is ready for QA

qasumitbagthariya commented:
### QA Update ⚠️
---

In the cart section, the "Coupon code" label and the input field next to it are slightly misaligned. 

![image](https://github.com/user-attachments/assets/eeb4f1ac-977a-42b2-b608-ec9c441ea00f)

![image](https://github.com/user-attachments/assets/e5ed96da-5f91-4936-b048-31d8dade81b1)

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis committed:
Fix label misalignment

Manussakis commented:
@dkotter I have updated the label's left position. Please review it.
Thanks!

- [Fix label misalignment](https://github.com/woocommerce/woocommerce/pull/52272/commits/97d846c63809658c4ab70e6754bbbd109f62cae7)

qasumitbagthariya commented:
@Manussakis 

In this branch, I notice following points

1) The "Apply coupon" button is not aligned.
Theme: Storefront
Page: Checkout

![image](https://github.com/user-attachments/assets/dcf920b8-892d-4db5-9a9e-54522dd19417)

2) The coupon code does not exist message not align
Theme: Storefront
Page: cart

![image](https://github.com/user-attachments/assets/c277e063-f543-4224-a074-b9e6473c1706)

---

⚠️ [Question: 3](https://github.com/woocommerce/woocommerce/pull/52272#issuecomment-2442123299),I noticed that the 'Coupon code' label and input field in the cart section are slightly misaligned. Should we adjust the font size, center the elements, or leave it as is?"

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis commented:
@qasumitbagthariya 

> 1. The "Apply coupon" button is not aligned.

This is expected as flagged in the PR description.

> 2. The coupon code does not exist message not align

This is out of the scope of this task. If you perform the same test in the trunk branch you will get the same result.
Storefront doesn't use Woo core stylesheet and this error message will be misaligned until the theme gets updated by the Storefront team. 😢

> 3. I noticed that the 'Coupon code' label and input field in the cart section are slightly misaligned. Should we adjust the font size, center the elements, or leave it as is?"

I have fixed that under this commit: [Fix label misalignment](https://github.com/woocommerce/woocommerce/pull/52272/commits/97d846c63809658c4ab70e6754bbbd109f62cae7). Did you pull the latest?

![image](https://github.com/user-attachments/assets/69ece84b-be4b-480d-857b-fdcd01128549)

qasumitbagthariya commented:
Thanks for the detailed comment @Manussakis Approving this PR 👍

vikrampm1 commented:
@wavvves this PR is ready for your review!

vikrampm1 commented:
@wavvves noting that this PR is awaiting your review!

wavvves reviewed (changes requested):
Hi @vikrampm1 @Manussakis, apologies for the delay—I missed that this was marked ready for review (keeping PRs in draft while WIP can help avoid confusion).

Could you clarify the reasoning behind introducing these regressions in Storefront (and possibly other themes)? Are there parallel efforts in Storefront to address these? If so, is this PR dependent on those? 
Alternatively, were these regressions communicated and deemed acceptable by the core/Storefront teams?

I might be overcautious, but I can’t approve this without clarity.
As an alternative, have you considered a block-based solution with labels styled like placeholders that float on focus (as seen in [Material Design](https://m2.material.io/components/text-fields))?

![image](https://github.com/user-attachments/assets/e4af115d-e464-4e47-8507-493d245152ca)

nerrad commented:
Besides the feedback @wavvves has added, these changes look like they will impact _more_ than just Storefront (and potentially multiple themes). Questions/feedback:

- What themes have been tested with these changes? At a minimum this should be tested across all WP core themes and ideally a few themes that explicitly support themes in the ecosystem (Astra). We need to know the impact of these changes as a way of influencing any decisions about merging the PR (or potential approaches).
- What exploration has been done in improving accessibility in a way that _doesn't_ impact existing styles/implementations that have been done? We should work on any potential approaches there before shipping something this breaking (how "breaking" is dependant on the impact known from the previous point).
- Even if we decide potential breaking change is warranted because of the accessibility improvements, we need to make sure there is plan for notifying the ecosystem before shipping. We _may_ even want to ship the change with theme-specific fixes in Woo core if necessary to minimize the impact to stores (which IMO is a last resort, but still something that should be considered).

nerrad commented:
Also cc @elizaan36 since this impacts design.

Manussakis commented:
@wavvves @nerrad  Thanks for your feedback!

> Could you clarify the reasoning behind introducing these regressions in Storefront (and possibly other themes)?

This feature relies mainly on two changes:
- Removing the `scree-reader-text` class from labels so they can be visible.
- Adding the `.is-active` class to labels when the associated input is focused so the labels get the floating transition.

The regression is introduced since the Storefront (and possibly other themes) doesn't load the Woo core stylesheet.

---

> Are there parallel efforts in Storefront to address these? If so, is this PR dependent on those?
Alternatively, were these regressions communicated and deemed acceptable by the core/Storefront teams?

I didn't start any communication with the Storefront team (or any other theme's team) since the changes introduced by this PR are still being reviewed. Once we reach an agreement on a better approach I will be happy to conduct this conversation if needed.

---

> As an alternative, have you considered a block-based solution with labels styled like placeholders that float on focus (as seen in [Material Design](https://m2.material.io/components/text-fields))?

The problem only affects the classic variation of the Cart and Checkout pages. The Material Design solution you suggested seems to be exactly the same solution as I implemented. Unless I missed something.  🤔

---

@nerrad 

> What themes have been tested with these changes? At a minimum this should be tested across all WP core themes and ideally a few themes that explicitly support themes in the ecosystem (Astra). We need to know the impact of these changes as a way of influencing any decisions about merging the PR (or potential approaches).

I tested all WP core themes (from Twenty Seventeen to Twenty Twenty-Four) as well as Storefront.
The impact is layouts becoming off as a result of removing the `scree-reader-text` from `labels` as they now occupy space on the page. Themes that don't load the Woo core stylesheet are likely to be affected. Other themes may need minor adjustments to position the labels.

---

> What exploration has been done in improving accessibility in a way that doesn't impact existing styles/implementations that have been done? We should work on any potential approaches there before shipping something this breaking (how "breaking" is dependant on the impact known from the previous point).

If we want to meet the [Label in Name (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html) WCAG criteria, the labels must be visible. This change is enough to cause regression on other themes, as explained above. My effort to give them the floating behavior was aimed at:
- Minimizing regressions at least for themes that use the Woo core stylesheet.
- Sticking to the floating behavior already in place on block-based pages.

---

> Even if we decide potential breaking change is warranted because of the accessibility improvements, we need to make sure there is plan for notifying the ecosystem before shipping. We may even want to ship the change with theme-specific fixes in Woo core if necessary to minimize the impact to stores (which IMO is a last resort, but still something that should be considered).

I am happy to start a conversation with the ecosystem if necessary once we decide the best approach to follow.

---

Considering the justifiable concern about shipping this PR, I am going back to the drawing board to see if I can come up with a less impactful solution. I will ping you once I have information to share.

Thanks again!

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis commented:
@wavvves @nerrad 

After further investigation of alternatives to fix this issue, I concluded that there is no way to prevent regressions on other themes. The main problem is that labels are hidden, and the first change to fix that is removing the `screen-reader-text` class from them. This simple change is sufficient to cause the regressions because themes are expecting the labels to be hidden.

Any other alternative that doesn't make labels visible won't meet the [Label in Name (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html) WCAG criteria, therefore, won't fix this issue.

Please, let me know your thoughts or if we need to involve more people before making a decision.

@elizaan36 Do you have any thoughts design-wise?

Once we decide, I can start the conversation with the themes' team if needed.

elizaan36 commented:
hey @Manussakis Thanks so much for taking a look at this issue and putting together a proposal to improve accessibility at checkout. I'd love to dive in and test the proposal to provide design feedback but it will take a little more time. Thank you for your patience!

> We may even want to ship the change with theme-specific fixes in Woo core if necessary to minimize the impact to stores (which IMO is a last resort, but still something that should be considered).

I would echo @nerrad 's point here. Would it make sense to roll this out only for the Storefront theme to ensure compatibility? Does this impact Cart and Checkout on legacy (classic) themes? I see the title of the issue only refers to Cart but I imagine the issue is broader than that.

Manussakis commented:
Hey, @elizaan36 👋

> Would it make sense to roll this out only for the Storefront theme to ensure compatibility?

The regression is for any theme that doesn't load the Woo core stylesheet.

---

> Does this impact Cart and Checkout on legacy (classic) themes? I see the title of the issue only refers to Cart but I imagine the issue is broader than that.

This PR fixes the issue on the Coupon form on the classic Cart/Checkout and the Shipping address form on the classic Cart.

---

Another way to address this is by creating a setting that enables/disables the feature introduced by this PR (floating label). This way, theme teams could play with it and adjust their themes accordingly, instead of coming across a change in the layout first.

vikrampm1 commented:
@elizaan36 @wavvves any further thoughts any the above as that would unblock us from proceeding the PR further?

Manussakis committed:
Merge branch 'trunk' into fix/51900

elizaan36 commented:
Thanks again for the efforts here. I think the misalignments introduced in the PR are severe enough that we shouldn't ship the changes as-is. (Examples)

<img width="405" alt="image" src="https://github.com/user-attachments/assets/24362101-421a-436f-9073-cfa4906c14a0" />
<img width="402" alt="image" src="https://github.com/user-attachments/assets/facfbe3f-f70e-47d6-90fd-8eb3a22e6387" />

It's important to improve the accessibility of these forms and I like the idea of applying the same "float" approach as blocks, but I'd lean towards having parallel work in top Woo themes to ensure compatibility before. If we introduce the update behind a setting that merchants have access to, this adds another layer of complexity to an already complex admin/store editing experience. I'd lean towards reaching the best defaults possible and working with the classic theme builders in parallel if possible to ensure compatibility.

Some of the confusion on the approach could be caused by the before and after pictures in the PR description, because the "After" pictures don't show the floating placeholders on focus. They show labels outside the form with a new (*) required indication. It would speed things up to have the "After" photos represent what the shopper will see for the top themes. Besides the WP core themes, Storefront, and Astra, we should consider Hello Elementor, Divi, and Flatsome, which are all top classic themes.

Looping in @pierorocca for awareness on the proposed update.

wavvves commented:
Well, I believe we're all now agreeing here on three things:
- Persistent labels are mandatory (per WCAG guidelines).
- The solution from blocks, floating labels, is the most evident choice. Any other choice would lead to similar problems.
- Breaking compatibility is not an acceptable tradeoff. For example, accepting to roll this out without 100% coverage initially would be a much more acceptable tradeoff. --> Seems everyone is here at this point.

Taking this consideration, I propose we quickly discuss the PR's approach, and have a look, from code perspective if we have any angles to explore in a last attempt to ship floating labels. I don't have a definitive answer, but I can bet 2 cents to start with.

Since this is going to be addressing elements present in templates (`plugins/woocommerce/templates/cart/shipping-calculator.php` and `plugins/woocommerce/templates/cart/cart.php`), and those are versioned, we might be able to leverage that to ship this in a controlled way, ensuring not breaking styles at the sacrifice of coverage.

My initial thought would be to attempt to encapsulate styles and scripts within the templates to implement the floating labels.
- On the plus side, this would allow avoiding shipping CSS/JS that alters unpredictable elements. Theme developers could leverage the warning about outdated templates to update sensibly.
- On the downside, it might not be the ideal way to deliver CSS/JS, and we might have to be _smart_ about it.

This could be hardened by introducing wrapper element classes, and having conditional changes done by JS in order to gracefully fallback themes that don't implement these templates and for which we might want to keep things as is.

Another thing we can leverage, is that we know exactly what we're targeting here, with the shipping calculator and coupon input (it's on the PR title. I'm not sure at this point what filters can mess with those, but this is different than targeting all input fields blindly. With that, I sense that we might get the opportunity to be a bit authoritative about margins and alignments.

One thing, I would like to add, is that addressing this with big themes in mind might be counterintuitive. These are the ones we probably expect to pick up any new template change quickly. I have bigger doubts about the vast usage of free and no longer supported themes.

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis commented:
@pierorocca 

> On part 2, I'm going to have to come back to this next week so that I can give it the attention it deserve

Thanks!

---

> Android Talkback the first thing voiced for me is the Coupon input field. I had to navigate backward to get to the instructive copy.

This is expected, once the form is open by clicking on the link, the coupon input is programmatically focused on, that's why it's voiced first.

---

> I agree the descriptive text should have the aria-describedby while the error should use aria-errormessage and aria-invalid.

The [screen reader support for `aria-errormessage` is short](https://www.digitala11y.com/aria-errormessageproperties/). Even the [W3C example of form errors](https://design-system.w3.org/styles/form-errors.html) uses `aria-describedby` instead.
 We can continue using `aria-describedby` since this attribute supports multiple message ids. I tested it locally and it worked. I can implement this improvement once we get the final direction on this issue as a whole.

Thanks!

wavvves commented:
Changing the PR state back to draft temporarily.

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis committed:
Associate coupon description to its input on legacy Checkout page

Manussakis committed:
Remove coupon float label style from mixins

Manussakis committed:
Remove coupon float label style from T19

Manussakis committed:
Remove coupon float label style from T17

Manussakis committed:
Remove coupon float label style from T21

Manussakis committed:
Remove coupon float label style from T23

Manussakis committed:
Remove coupon float label style from T22

Manussakis committed:
Remove coupon float label style from TT

Manussakis committed:
Remove coupon float label style from woocommerce core stylesheet

Manussakis committed:
Revert float label functionality from coupon and shipping address forms on legacy Cart and Checkout pages

Manussakis committed:
Add empty line

Manussakis committed:
Revert change in coupon test

Manussakis committed:
Hide coupon label again

Manussakis commented:
@pierorocca After our discussion I have done the following:

- Reverted the changes to remove the floating label feature.
- Associated the coupon description with its input on the classic Checkout page.
- Made the shipping address form labels visible on the classic Cart page.
- Updated the PR's title and description.

@nerrad @elizaan36 what are your thoughts after these updates?

Manussakis commented:
@dkotter This PR is ready for another round of code review.
Thanks!

nerrad commented on plugins/woocommerce/client/legacy/js/frontend/checkout.js:
```diff
@@ -685,14 +685,14 @@ jQuery( function( $ ) {
 				.addClass( 'has-error' )
 				.attr( 'aria-invalid', 'true' )
-				.attr( 'aria-describedby', 'coupon-error-notice' );
+				.attr( 'aria-describedby', 'coupon-error-notice coupon-code-input-description' );
```
If a store has a template override that does not include this selector, this JS will break? This comment applies to the line change below as well.

nerrad replied:
Sorry I missed this was setting the attr.

elizaan36 commented:
Hey all, Just catching up and glad to see the progress here. 

It was my understanding from the [previous comments](https://github.com/woocommerce/woocommerce/pull/52272#issuecomment-2508390928) that we'd use the floating label (Material design pattern) for the field labels. It seems quite redundant to have the label both inside and above the form field. Will the label also appear above the form field on the Cart as shown here?

<img width="354" alt="Screenshot 2025-01-07 at 2 23 30 PM" src="https://github.com/user-attachments/assets/2b231e88-f119-4832-8ec8-e7e045f6acb1" />

I see we've resolved the Coupon form issues, and this is outside the scope of this issue, but I'm curious why the "Have a coupon" top banner notice is necessary on classic checkout especially since it repeats the message in the helper text almost exactly.

<img width="730" alt="Screenshot 2025-01-07 at 2 19 51 PM" src="https://github.com/user-attachments/assets/ee7320aa-ea4f-4116-8093-1b4fb104015a" />

Manussakis committed:
Merge branch 'trunk' into fix/51900

Manussakis committed:
Remove placeholders from shipping calculator template

Manussakis committed:
Revert changes in coupon form on classic checkout

Manussakis commented:
Hey @elizaan36 thanks for your feedback!

> It was my understanding from the https://github.com/woocommerce/woocommerce/pull/52272#issuecomment-2508390928 that we'd use the floating label (Material design pattern) for the field labels.

I believe you misunderstood. I finish that comment by saying:

> Considering the justifiable concern about shipping this PR, I am going back to the drawing board to see if I can come up with a less impactful solution.

The floating labels are causing too many regressions on themes that don't load the Woo core stylesheet.

---

>  It seems quite redundant to have the label both inside and above the form field. Will the label also appear above the form field on the Cart as shown here?

You are right, I removed the placeholders and only kept the labels above the fields.

![image](https://github.com/user-attachments/assets/f08da94f-3783-43cc-9dfe-3c19cd727ada)

---

> I see we've resolved the Coupon form issues, and this is outside the scope of this issue, but I'm curious why the "Have a coupon" top banner notice is necessary on classic checkout especially since it repeats the message in the helper text almost exactly.

I removed any fixes related to the coupon form from this PR and created the https://github.com/woocommerce/woocommerce/issues/54336 issue. Let's continue this discussion there 😉.

Thanks!

elizaan36 commented:
> The floating labels are causing too many regressions on themes that don't load the Woo core stylesheet.

Thanks for the clarification. Makes sense.

> You are right, I removed the placeholders and only kept the labels above the fields.

Nice, these changes look good to me!

Thanks for creating the new issue for the coupon form, I'll add a comment on there.

Manussakis commented:
Thanks @elizaan36 !

@qasumitbagthariya This PR is ready for another round of QA.
Thanks!

wavvves commented:
Hi @Manussakis is this ready for review? Please mark it as ready, I'll pick this up tomorrow 👍🏼

Manussakis commented:
> Hi @Manussakis is this ready for review? Please mark it as ready, I'll pick this up tomorrow 👍🏼

@wavvves Since I have updated the PR I sent it to another round of QA on the 10up side before sending it to WC review. That's why I haven't marked it as ready for review yet. Once it's done I will flag you and remove it from Draft.

Thanks!

wavvves commented:
> > Hi @Manussakis is this ready for review? Please mark it as ready, I'll pick this up tomorrow 👍🏼
> 
> 
> 
> @wavvves Since I have updated the PR I sent it to another round of QA on the 10up side before sending it to WC review. That's why I haven't marked it as ready for review yet. Once it's done I will flag you and remove it from Draft.
> 
> 
> 
> Thanks!

Thanks, I'll keep an eye out 😉

qasumitbagthariya commented:
QA Update: ✅
---
I have checked this PR in the `Manussakis:fix/51900` branch and changes working as expected.


The latest changes looks good to me

**storefront**
<img width="1480" alt="image" src="https://github.com/user-attachments/assets/4d13d668-2030-4c2e-a185-4b7fba9267a6" />

**TT4**
<img width="811" alt="image" src="https://github.com/user-attachments/assets/cafc887c-67dd-4948-a35a-249695ee2fe4" />


**TT5**
<img width="951" alt="image" src="https://github.com/user-attachments/assets/e3f66d13-3982-4c50-851d-254e195bd238" />


@Manussakis 

Misaligned Coupon Code Section

As per the screenshot, the Coupon Code text box and the Apply Coupon button are not properly aligned. Additionally, the width of the text box appears too small, which could affect user experience.

 Do we need to address these alignment and size as part of this PR, or should we keep it as it is for now ?

![image](https://github.com/user-attachments/assets/2ef69d4d-311c-4293-a66f-a0bb156a0b67)


cc @vikrampm1

Manussakis commented:
Thanks, @qasumitbagthariya !

> As per the screenshot, the Coupon Code text box and the Apply Coupon button are not properly aligned. Additionally, the width of the text box appears too small, which could affect user experience.

Any updates in the coupon form have been reverted. The issue you flagged is out of the scope of this PR.

Thanks!

Manussakis commented:
@wavvves This PR is ready for review.
Thanks!

Manussakis commented:
Hey @pierorocca 
Regarding @elizaan36's callout, I have already created the https://github.com/woocommerce/woocommerce/issues/54336 issue. 😉 

cc: @qasumitbagthariya

wavvves reviewed (approved):
Hi @Manussakis, thank you so much for the resilience in bringing this forward 💪🏼 
I have tested the PR against mentioned themes and a few others, and didn't spot any severe breakage (just the occasional ~2px margin tweak). Since the changes are scoped to a template there should be no issue 👍🏼 
Approving ✅

wavvves commented:
> @wavvves sounds great, so otherwise good to merge here or is there anything else needed from Gabriel?

No, I'm merging this right now.
@retrofox I don't think there's anything to follow-up on storefront after this 👍🏼

wavvves commented:
> As per the screenshot, the Coupon Code text box and the Apply Coupon button are not properly aligned. Additionally, the width of the text box appears too small, which could affect user experience.

Possible relation with https://github.com/woocommerce/woocommerce/issues/54062
